Content.Server/GameTicking/GameTicker.Spawning.cs

```diff
@@ -172,23 +172,24 @@ private void SpawnPlayer(IPlayerSession player, HumanoidCharacterProfile charact
 
             _playTimeTrackings.PlayerRolesChanged(player);
 
+
+            var mobMaybe = _stationSpawning.SpawnPlayerCharacterOnStation(station, job, character);
+            DebugTools.AssertNotNull(mobMaybe);
+            var mob = mobMaybe!.Value;
+
+            newMind.TransferTo(mob);
+
             if (lateJoin)
             {
                 _chatSystem.DispatchStationAnnouncement(station,
                     Loc.GetString(
                         "latejoin-arrival-announcement",
-                    ("character", character.Name),
+                    ("character", MetaData(mob).EntityName),
```

Comment:
Use Identity to get the name.

Reply:
couldn't this recreate the "Unknown has joined the station" bugged announcement from unity

Reply:
> couldn't this recreate the "Unknown has joined the station" bugged announcement from unity

With Entity() no.